app/models/package_manager/go.rb

```diff
@@ -128,7 +130,18 @@ def self.project(name)
       # base package manager handles if the project is not present
       return nil if doc_html.text.blank?
 
-      { name: name, html: doc_html, overview_html: doc_html }
+      raw_project = { name: name, html: doc_html, overview_html: doc_html }
+
+      # pages on pkg.go.dev can be categorized as 'package', 'module', 'command', or 'directory'. We only scrape Go Modules.
+      unless module_type?(raw_project: raw_project)
```

Comment:
WDYT about wrapping up this `raw_project` analysis into a separate object, rather than creating more public methods to process passed-in state on this class?

Reply:
yeah, that might make sense, or I was also thinking of parameterizing it, e.g. `has_type?("module", raw_project: raw_project)`. I'll play around with that now